sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisClientTestBase.java

```diff
@@ -0,0 +1,605 @@
+    }
+
+    private static void validateObjects(ObjectsResult objectsResult) {
+        assertNotNull(objectsResult);
+        assertTrue(objectsResult.getValues().size() > 1);
+
+        Boolean found1 = false;
+        for (DetectedObject object : objectsResult.getValues()) {
+            assertNotNull(object);
+            assertNotNull(object.getTags());
+            assertEquals(object.getTags().size(), 1);
+            DetectedTag tag = object.getTags().get(0);
+            assertNotNull(tag);
+            assertNotNull(tag.getName());
+            assertFalse(tag.getName().isEmpty());
+            assertTrue(tag.getConfidence() > 0.0);
+            assertTrue(tag.getConfidence() < 1.0);
+            // We expect to see this in the list of objects
+            if (tag.getName().toLowerCase().equals("person")) {
+                found1 = true;
+            }
+        }
+        assertTrue(found1);
+
+        // Make sure each object box is unique
+        for (int i = 0; i < objectsResult.getValues().size(); i++) {
+            for (int j = i + 1; j < objectsResult.getValues().size(); j++) {
+                ImageBoundingBox boxI = objectsResult.getValues().get(i).getBoundingBox();
+                ImageBoundingBox boxJ = objectsResult.getValues().get(j).getBoundingBox();
+                assertFalse(
+                    boxI.getX() == boxJ.getX()
+                    && boxI.getY() == boxJ.getY()
+                    && boxI.getHeight() == boxJ.getHeight()
+                    && boxI.getWidth() == boxJ.getWidth());
+            }
+        }
+    }
+
+    private static void validateTags(ImageAnalysisResult result) {
+        TagsResult tags = result.getTags();
+        validateTags(tags);
+    }
+
+    private static void validateTags(TagsResult tagsResult) {
+        assertNotNull(tagsResult);
+        assertNotNull(tagsResult.getValues());
+        assertTrue(tagsResult.getValues().size() > 1);
+
+        Boolean found1 = false, found2 = false;
+        for (DetectedTag tag : tagsResult.getValues()) {
+            assertNotNull(tag.getName());
+            assertFalse(tag.getName().isEmpty());
+            assertTrue(tag.getConfidence() > 0.0);
+            assertTrue(tag.getConfidence() < 1.0);
+            // We expect to see both of these in the list of tags
+            if (tag.getName().toLowerCase().equals("person")) {
+                found1 = true;
+            }
+            if (tag.getName().toLowerCase().equals("laptop")) {
+                found2 = true;
+            }
+        }
+        assertTrue(found1);
+        assertTrue(found2);
+
+        // Make sure each tag is unique
+        for (int i = 0; i < tagsResult.getValues().size(); i++) {
+            for (int j = i + 1; j < tagsResult.getValues().size(); j++) {
+                assertFalse(
+                    tagsResult.getValues().get(j).getName().equals(
+                    tagsResult.getValues().get(i).getName()));
+            }
```

Comment:
Does this test make sense? This is comparing object types without an `equals` overload, which unless they are pointing to the same object will always be false.

Reply:
@alzimmermsft The `.getName()` method here returns a regular Java `String` object. I need to make sure that every tag name (a String) in the list of returned image tags is unique. So I check all pairs and compare two Java strings, with the built-in String.equals() method. I double checked and the code should be good as is. Please reach out to me over IM if you still have comments about this. Thank you!

Reply:
Oh sorry, missed the `getName` call, this makes more sense now

Reply:
@alzimmermsft good point, I switched to using assertNotEquals. Thanks!